python/dgl/data/tu.py

```diff
@@ -3,50 +3,48 @@
 import os
 import random
 
-from .utils import download, extract_archive, get_download_dir, loadtxt
+
+from .dgl_dataset import DGLBuiltinDataset
+from .utils import loadtxt, save_graphs, load_graphs, save_info, load_info
+from .. import backend as F
 from ..utils import retry_method_with_fix
 from ..graph import DGLGraph
 
-class LegacyTUDataset(object):
-    """
+class LegacyTUDataset(DGLBuiltinDataset):
+    r"""
     TUDataset contains lots of graph kernel datasets for graph classification.
     Use provided node feature by default. If no feature provided, use one-hot node label instead.
     If neither labels provided, use constant for node feature.
 
-    :param name: Dataset Name, such as `ENZYMES`, `DD`, `COLLAB`
-    :param use_pandas: Default: False.
+    Parameters
+    ----------
+    name : str
+        Dataset Name, such as `ENZYMES`, `DD`, `COLLAB`
+    use_pandas : bool
         Numpy's file read function has performance issue when file is large,
         using pandas can be faster.
-    :param hidden_size: Default 10. Some dataset doesn't contain features.
+        Default: False
+    hidden_size : int
+        Some dataset doesn't contain features.
         Use constant node features initialization instead, with hidden size as `hidden_size`.
+        Default : 10
+    max_allow_node : int
+        Remove graphs that contains more nodes than `max_allow_node`.
+        Default : None
 
     """
 
     _url = r"https://ls11-www.cs.tu-dortmund.de/people/morris/graphkerneldatasets/{}.zip"
```

Comment:
1. Instead of setting _url as class attribute, setting it as a local constant.
2. The dataset is downloaded from the outside, need SHA1 check.